libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -108,13 +108,17 @@ void HarmonicNotchFilter<T>::create(uint8_t harmonics)
             _num_filters++;
         }
     }
-    _filters = new NotchFilter<T>[_num_filters];
+    if (_num_filters > 0) {
+        _filters = new NotchFilter<T>[_num_filters];
+    }
     _harmonics = harmonics;
 }
 
 template <class T>
 HarmonicNotchFilter<T>::~HarmonicNotchFilter() {
-    delete[] _filters;
+    if (_num_filters > 0) {
```

Comment:
This check isn't needed, it's safe to delete a nullptr variable.

Reply:
True, but unless you feel very strongly about this I prefer the explicitness.

Reply:
I agree with oxinarf actually that just do the delete and maybe also set the _num_filters = 0 if that makes sense.